meilisearch/index.py

```diff
@@ -1530,8 +1543,13 @@ def __settings_url_for(self, sub_route: str) -> str:
     def _build_url(
         self,
         primary_key: Optional[str] = None,
+        csv_delimiter: Optional[str] = None,
     ) -> str:
-        if primary_key is None:
+        parameters = {}
+        if primary_key:
+            parameters["primaryKey"] = primary_key
+        if csv_delimiter:
+            parameters["csvDelimiter"] = csv_delimiter
+        if primary_key is None and csv_delimiter is None:
             return f"{self.config.paths.index}/{self.uid}/{self.config.paths.document}"
-        primary_key = parse.urlencode({"primaryKey": primary_key})
-        return f"{self.config.paths.index}/{self.uid}/{self.config.paths.document}?{primary_key}"
+        return f"{self.config.paths.index}/{self.uid}/{self.config.paths.document}?{parse.urlencode(parameters)}"
```

Comment:
This is what is specified in the specifications:
```
- A new query parameter, `csvDelimiter`, has been introduced to customize the csv delimiter used in the document. It can change between two `documentAddition`.
```

Reply:
OK seems weird to me to do it this way, but looks like it is correct :smile: